sky/dag.py

```diff
@@ -2,74 +2,253 @@
 import pprint
 import threading
 import typing
-from typing import List, Optional
+from typing import Dict, List, Optional, Set, Union
+
+import networkx as nx
 
 if typing.TYPE_CHECKING:
     from sky import task
 
+TaskOrName = Union['task.Task', str]
+
 
 class Dag:
     """Dag: a user application, represented as a DAG of Tasks.
 
+    This class allows users to define and manage directed acyclic graphs
+    (DAGs) of tasks, representing complex workflows or pipelines.
+
     Examples:
         >>> import sky
-        >>> with sky.Dag() as dag:
-        >>>     task = sky.Task(...)
+        >>> with sky.Dag(name='my_pipeline') as dag:
+        >>>     task1 = sky.Task(name='task1', ...)
+        >>>     task2 = sky.Task(name='task2', ...)
+        >>>     task1 >> task2
     """
 
-    def __init__(self) -> None:
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        tasks: Optional[List['task.Task']] = None,
+        dependencies: Optional[Dict[TaskOrName, Union[List[TaskOrName],
+                                                      TaskOrName]]] = None
```

Comment:
I dont feels like we should pass the dependencies as an argument - we already have the `>>` operator to specify this. Multiple ways to do the same thing would be confusing. Same for the tasks argument